### PR TITLE
Simplify inspector.

### LINKED ISSUE #5560: Simplify the JupyterLab inspector.
Following some conversations with @BoPeng and @ellisonbg, we should make JupyterLab's inspector do fewer things and be a single-purpose extension. In addition, we should provide an extension point so that the actual *content* of the inspector can be generated by a third-party extension. This functionality and API should be solidified before 1.0:

* The `remembers` functionality  of the inspector, which has lain dormant, should be removed.
  * The toolbar for the inspector should removed as it is the main UI manifestation of the `remembers` feature.
  * The internal list of previously inspected items that are kept in memory should be removed. The inspector should basically be a simple widget that displays the currently inspected thing only.
* The concept of multiple inspectors each with their own tabs should be reduced to a single inspector. A tab bar inside the tabbed interface of JupyterLab is not a great user experience.
* The logic that transforms a MIME bundle into a widget that gets displayed inside the inspector should be moved into a standalone `JupyterLabPlugin` in order to allow third-party extension to override it and generate their own widgets to show inside the inspector.
* The concept of *types* of inspector content should probably be removed as well.

cf. https://github.com/jupyterlab/jupyterlab/pull/4879, https://github.com/jupyterlab/jupyterlab/pull/5530

*Upshot:* What this means is that the only extension points of the inspector should be:

1. Intercepting the MIME bundle and returning custom widgets to render in the inspector panel instead of using the core renderer that ships with JupyterLab.
1. Replacing the inspector altogether.

For other similar functionality within a third-party extension, it probably makes most sense to simply create a main area widget or a side area widget that ships with the extension.

### PR DESCRIPTION
**Pull request by afshin**
* Fixes #5560 

* Fixes broken conda docs links.

* Adds inspector command to file menu and launcher, including a caption and icon.

* Makes inspector dependencies on palette, launcher, and restorer optional.

* Adds optional captions for launcher cards.

screen shot 2019-01-23 at 13 48 46

### CODE REVIEW

File: packages/inspector-extension/src/index.ts [hunk 1]
```diff
@@ -56,55 +53,52 @@ const inspector: JupyterLabPlugin<IInspector> = {
...
     }
 
+    // Add command to registry and palette.
+    commands.addCommand(command, {
+      label,
+      execute: () => {
+        createInspector();
+      }
+    });
+    palette.addItem({ command, category });
```
Comment (ian-r-rose) [inline packages/inspector-extension/src/index.ts]: I wonder if also adding this to the launcher and file menus would make it more discoverable. I know that lots of people are unaware that the inspector exists.

File: packages/inspector-extension/src/index.ts [hunk 2]
```diff
@@ -56,55 +53,52 @@ const inspector: JupyterLabPlugin<IInspector> = {
...
     });
 
-    // Add command to registry and palette.
-    commands.addCommand(command, {
-      label,
-      execute: () => {
-        if (!manager.inspector || manager.inspector.isDisposed) {
-          manager.inspector = newInspectorPanel();
+    // Create a proxy to pass the `source` to the current inspector.
+    const proxy: IInspector = Object.defineProperty({}, 'source', {
```
Comment (ian-r-rose) [inline packages/inspector-extension/src/index.ts]: Clever.

File: packages/inspector/style/index.css [hunk 3]
```diff
@@ -22,82 +22,19 @@
...
-  content: '\f060'; /* arrow-left */
-}
-
-.jp-InspectorItem .jp-Toolbar-button.jp-InspectorItem-forward::before {
-  content: '\f061'; /* arrow-right */
-}
-
-.jp-InspectorItem pre {
+.jp-Inspector-content pre {
   white-space: pre-wrap;
```
Comment (ian-r-rose) [inline packages/inspector/style/index.css]: Overflow scrolling and `pre-wrap` for the inspector appears to be broken in my testing.

File: packages/inspector-extension/src/index.ts (hunk 1 above)
Comment (afshin) [reply packages/inspector-extension/src/index.ts]: Good point. I added it to  the "File" menu and to the launcher. (**screenshot** in main PR description)

File: packages/inspector/style/index.css (hunk 3 above)
Comment (afshin) [reply packages/inspector/style/index.css]: Fixed here: https://github.com/jupyterlab/jupyterlab/pull/5776/commits/5e3316d27f579110c35e7983042f84d8a0540fcb

**Comment by jasongrout** [comment]
> trigger inspection when the cursor moves, or when a new cell/editor is focused

+1. I've wished for these too.

**Comment by ian-r-rose** [comment]
Okay, I'm merging this, and will file a new issue for cursors/focusing.

**Comment by ian-r-rose** [comment]
Thanks @afshin, I like that line delta.

**Pull request merged**